op-service/eth/errors.go

```diff
@@ -0,0 +1,22 @@
+package eth
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/ethereum/go-ethereum"
+)
+
+// MaybeAsNotFoundErr checks if the error is an ethereum.NotFound error
+// or has an error string that heuristically indicates that it is this error.
+// If so, it returns ethereum.NotFound, otherwise it returns the original error.
```

Comment:
Also worth adding a comment that on RPC, following the standard ethereum execution-api specs and test vectors, a block-by-hash request is expected to return a null result and no error code/string, if really not found. It's technically not valid execution-client behavior when it returns an error instead of matching the specified null behavior. So users of the RPC still need to check for a nil result when interacting with the raw JSON RPC client.

Reply:
added to the doc.